web/renderer/components/pageComponents/ConnectionsPage/NewConnection/CloneDoltDatabaseForm.tsx

```diff
@@ -0,0 +1,157 @@
+    onCloneDoltHubDatabase,
+  } = useConfigContext();
+
+  const { disabled, message } = getStartLocalDoltServerDisabled(
+    state,
+    storedConnections,
+  );
+
+  return (
+    <>
+      <FormInput
+        value={state.owner}
+        onChangeString={owner => {
+          setState({ owner });
+          setErr(undefined);
+        }}
+        label="Owner Name"
+        labelClassName={css.label}
+        placeholder="e.g. dolthub"
+        light
+      />
+      <FormInput
+        value={state.database}
+        onChangeString={n => {
+          setState({ database: n });
+          setErr(undefined);
+        }}
+        label="Remote Database Name"
+        labelClassName={css.label}
+        placeholder="e.g. my-database (required)"
+        light
+      />
+      <FormInput
+        value={state.name}
+        onChangeString={n => {
+          setState({ name: n });
+          setErr(undefined);
+        }}
+        label="Connection Name"
+        labelClassName={css.label}
+        placeholder="e.g. my-database (required)"
+        light
+      />
+      <FormInput
+        label="Port"
+        value={state.port}
+        onChangeString={p => {
+          setState({ port: p });
+          setErr(undefined);
+        }}
+        placeholder="e.g. 3658 (required)"
+        light
+        labelClassName={css.label}
+      />
+      <ButtonsWithError error={error} className={css.buttons}>
+        <Popup
+          position="bottom center"
+          on={["hover"]}
+          contentStyle={{
+            fontSize: "0.875rem",
+            width: "fit-content",
+          }}
+          closeOnDocumentClick
+          trigger={
+            <div>
+              {state.loading ? (
+                <div className={css.cloneProgress}>
+                  <span>Cloning..</span>
+                  <div className={css.progressContainer}>
+                    <div
+                      className={css.progressBar}
+                      style={{ transform: `scaleX(${state.progress / 100})` }}
+                    />
+                  </div>
+                </div>
+              ) : (
+                <Button
+                  type="submit"
+                  disabled={disabled || state.loading}
+                  className={css.button}
+                  onClick={onCloneDoltHubDatabase}
+                >
+                  Start Clone
+                </Button>
+              )}
+            </div>
+          }
+          disabled={!disabled}
+        >
+          {disabled && <div className={css.popup}>{message}</div>}
+        </Popup>
+      </ButtonsWithError>
+    </>
+  );
+}
+
+type DisabledReturnType = {
+  disabled: boolean;
+  message?: React.ReactNode;
+};
+
+function getStartLocalDoltServerDisabled(
+  state: ConfigState,
+  connections?: DatabaseConnectionFragment[],
+): DisabledReturnType {
+  const disabled =
+    !state.name ||
+    !state.owner ||
+    !!connections?.some(connection => connection.isLocalDolt);
```

Comment:
You don't really need this since you go through each of these conditions below? Can just specify `disabled: true` or `disabled: false` at the end